File: from-0501-to-0600/0515-find-largest-value-in-each-tree-row/solution.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func largestValues(root *TreeNode) []int {
    result := []int{}
    q := []*TreeNode{}
    if root != nil {
        q = append(q, root)
    }

    for len(q) > 0 {
        lenRow := len(q)
        rowMax := math.MinInt
        for i := 0; i < lenRow; i++ {
            node := q[0]
            q = q[1:]
            if node.Val > rowMax {
                rowMax = node.Val
            }
            if node.Left != nil {
                q = append(q, node.Left)
            }
            if node.Right != nil {
                q = append(q, node.Right)
            }
        }
        result = append(result, rowMax)
    }

    return result
}

File: from-0501-to-0600/0515-find-largest-value-in-each-tree-row/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def largestValues(self, root: Optional[TreeNode]) -> List[int]:
        q = deque()
        result = []
        if root:
            q.append(root)

        while q:
            n = len(q)
            rowMax = float('-inf')
            for _ in range(n):
                node = q.popleft()
                rowMax = max(rowMax, node.val)

                if node.left:
                    q.append(node.left)
                if node.right:
                    q.append(node.right)

            result.append(rowMax)
        return result